Assets/HoloToolkit/Input/Scripts/InputSources/GesturesInput.cs

```diff
@@ -89,80 +93,120 @@ protected override void OnDestroy()
 
                 gestureRecognizer.Dispose();
             }
+            if (navigationGestureRecognizer != null)
+            {
+                navigationGestureRecognizer.NavigationStartedEvent -= OnNavigationStartedEvent;
+                navigationGestureRecognizer.NavigationUpdatedEvent -= OnNavigationUpdatedEvent;
+                navigationGestureRecognizer.NavigationCompletedEvent -= OnNavigationCompletedEvent;
+                navigationGestureRecognizer.NavigationCanceledEvent -= OnNavigationCanceledEvent;
+
+                navigationGestureRecognizer.Dispose();
+            }
+        }
 
-            base.OnDestroy();
+        protected virtual void OnDisable()
+        {
+            StopGestureRecognizer();
+        }
+
+        protected virtual void OnEnable()
+        {
+            if (RecognizerStart == RecognizerStartBehavior.AutoStart)
+            {
+                StartGestureRecognizer();
+            }
+        }
+
+        /// <summary>
+        /// Make sure the gesture recognizer is off, then start it.
+        /// Otherwise, leave it alone because it's already in the desired state.
+        /// </summary>
+        public void StartGestureRecognizer()
+        {
+            if (gestureRecognizer != null && !gestureRecognizer.IsCapturingGestures())
+            {
+                gestureRecognizer.StartCapturingGestures();
+            }
+            if (navigationGestureRecognizer != null && !navigationGestureRecognizer.IsCapturingGestures())
+            {
+                navigationGestureRecognizer.StartCapturingGestures();
+            }
+        }
+
+        /// <summary>
+        /// Make sure the gesture recognizer is on, then stop it.
+        /// Otherwise, leave it alone because it's already in the desired state.
+        /// </summary>
+        public void StopGestureRecognizer()
+        {
+            if (gestureRecognizer != null && gestureRecognizer.IsCapturingGestures())
+            {
+                gestureRecognizer.StopCapturingGestures();
+            }
+            if (navigationGestureRecognizer != null && navigationGestureRecognizer.IsCapturingGestures())
+            {
+                navigationGestureRecognizer.StopCapturingGestures();
+            }
         }
 
-        private void OnTappedEvent(InteractionSourceKind source, int tapCount, Ray headRay)
+        protected void OnTappedEvent(InteractionSourceKind source, int tapCount, Ray headRay)
         {
-            SourceClickEventArgs args = new SourceClickEventArgs(this, 0, tapCount);
-            RaiseSourceClickedEvent(args);
+            inputManager.RaiseSourceClicked(this, 0, tapCount);
```

Comment:
Because the input system forwards information to whatever is being gazed at, I don't think you need the head ray coming from the gesture itself. On top of that, if you somehow need to know the current stabilized head ray at any point in time, you can ask GazeManager.